File: Homeworks/01. C# Fundamentals/06. Loops/14. Hex to Decimal/HexToDecimal.cs
﻿using System;

class HexToDecimal
{
    static void Main()
    {
        string hexNumber = Console.ReadLine();

        long decimalNumber = 0;

        for (int i = 0; i < hexNumber.Length; i++)
        {
            int multiplier;

            switch (hexNumber[hexNumber.Length - i - 1])
            {
                case 'A': multiplier = 10; break;
                case 'B': multiplier = 11; break;
                case 'C': multiplier = 12; break;
                case 'D': multiplier = 13; break;
                case 'E': multiplier = 14; break;
                case 'F': multiplier = 15; break;
                default: multiplier = int.Parse(hexNumber[hexNumber.Length - i - 1].ToString()); break;
            }

            decimalNumber += multiplier * (long)Math.Pow(16, i);
        }

        Console.WriteLine(decimalNumber);

    }
}

File: Homeworks/06. JS Fundamentals/06. Loops/05. Hex to Decimal/HexToDecimal.js
function HexToDecimal(args) {
    var hex = args[0], decimal = 0, hexDigits = ['0', '1', '2', '3', '4', '5', '6', '7', '8', '9', 'A', 'B', 'C', 'D', 'E', 'F'];
    for(var i = hex.length - 1; i >= 0; i -= 1) {
        decimal += hexDigits.indexOf(hex[i]) * Math.pow(16, hex.length - i - 1);
    }
    
    console.log(decimal);
}